sktime/forecasting/theta.py

```diff
@@ -94,13 +94,17 @@ def __init__(self, initial_level=None, deseasonalize=True, sp=1):
 
         self.sp = sp
         self.deseasonalize = deseasonalize
-
+        self.initialization_method_ = "known" if initial_level else "estimated"
```

Comment:
This should happen in `fit` rather than in `__init__`.

Reply:
I am not sure, why? The super class requires this a parameter when initializing right. Should I override the intialization_method variable instead of passing it?

Sorry, I am just confused and want to make sure.

Reply:
Yes, we should probably change it in the super class then. The problem is that if someone uses `set_params(...)` to change hyper-parameters after construction, `__init__` isn't called again and hence the `initialization_method` wouldn't be updated correctly.

Reply:
Wouldn't that create the same problem for ExponentialSmoothing? Should I add it to the fit method of the super class (that might make the most sense)?

Reply:
Yes sounds good!

Reply:
@mloning So I was changing this and realized what will happen if someone explicitly passes initialization_method to ExponentialSmoothing (since it gives an option to that while ThetaModel doesn't). It should use what the user passed instead of what we need for ThetaModel. If I add it to the fit method of super class, that would always change the initialization_method, which isn't something we want.

In statsmodel, they do this in their ThetaModel and call ExponentialSmoothing with the correct parameter, which is what we are doing when we change it in __init__.

Now the most viable option would be to update the super class variable in fit method of ThetaModel. Let me know if I missed something here.

Reply:
Hi @yashlamba yes, makes sense, that should work if we set the variable before the call to the fit of the super class I think